node/src/protocol/consensus.rs

```diff
@@ -654,13 +675,24 @@ impl ConsensusProtocol for JoiningState {
 impl ConsensusProtocol for NodeState {
     async fn advance<C: ConsensusCtx + Send + Sync>(
         self,
-        ctx: C,
+        mut ctx: C,
         contract_state: ProtocolState,
     ) -> Result<NodeState, ConsensusError> {
         match self {
             NodeState::Starting => {
-                let node_data = ctx.secret_storage().load().await?;
-                Ok(NodeState::Started(StartedState(node_data)))
+                let persistent_node_data = ctx.secret_storage().load().await?;
+                let triple_storage = ctx.triple_storage();
+                let read_lock = triple_storage.read().await;
+                let triple_data_result = read_lock.load().await;
+                drop(read_lock);
+                let triple_data = match triple_data_result {
+                    Ok(vec_triple_data) => Some(vec_triple_data),
+                    _ => None,
+                };
```

Comment:
youu can shorten this to:
```suggestion
                let triple_data = match triple_data_result.ok();
```